controllers/machine_controller_phases.go

```diff
@@ -87,6 +87,12 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
 		return nil, err
 	}
 
+	// if external ref is paused, return error.
+	if util.IsPaused(cluster, obj) {
+		logger.V(3).Info("Infrastructure ref is paused")
+		return nil, errors.New("Infrastructure ref is paused")
```

Comment:
Should likely not return an error here, similar to the comment on the Cluster controller.